### PR TITLE
add hsb block, dict get, triggeredEvent, check value, inline script

### PR DESCRIPTION
**Pull request by stefan-hoehn**
Signed-off-by: Stefan Höhn <stefan@andreaundstefanhoehn.de>

Add a block that converts blockly’s rgb block with hexadecimal color representation to openhab HSB
Add a block that allows retrieving dictionary key / values
Add triggeredEvent to “contextual info” block
Add block to check if a stored value exists
Add block that allows to inline script

### CODE REVIEW

**Comment by ghys** [comment]
LGTM but please fix the formatting issues (see the Files tab).

File: bundles/org.openhab.ui/web/src/pages/settings/rules/script/blockly-editor.vue [hunk 1]
```diff
@@ -302,6 +309,7 @@
             </shadow>
           </value>
         </block>
+        <block type="oh_color_to_hsb" />
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/pages/settings/rules/script/blockly-editor.vue:312]: Can you add a colour_picker shadow block attached to the input?

File: bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-color.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  * Block
+  */
+  Blockly.Blocks['oh_color_to_hsb'] = {
+    init: function () {
+      this.appendValueInput('hexColor')
+        .appendField('hsb of')
+        .setCheck('Colour')
+      this.setInputsInline(false)
+      this.setOutput(true, 'String')
+      this.setColour(0)
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-color.js:19]: To keep the block the same color as the rest in this category (preferred):
```suggestion
      this.setColour('%{BKY_COLOUR_HUE}')
```
cf https://developers.google.com/blockly/guides/create-custom-blocks/block-colour#colour_references

Comment (stefan-hoehn) [reply bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-color.js:19]: Learned something new again - thx for the hint.

Comment (stefan-hoehn) [reply bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-color.js]: @ghys I committed this "this.setColour('%{BKY_COLOUR_HUE}')" here but I also want to add the shadow block (while you can work on the main branch, I have to do everything on the fork)

To not run into some git trouble again (I definitely want to learn that), what is the right approach?

Here's what I would do 

- fetch upstream  on my repo https://github.com/stefan-hoehn/openhab-webui on github
image
- should I do a "fetch upstream origin/main" locally as well?
- then "git pull" locally
- "git rebase upstream/main" locally 
- git commit of my shadow block (but that adds another commit, doesn't it?
- git push

**Comment by stefan-hoehn** [comment]
> LGTM but please fix the formatting issues (see the Files tab).

This is weird, I was 100% sure I ran npm link before :-(

Will do if course!

File: bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-color.js (hunk 2 above)
Comment (ghys) [reply bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-color.js]: 1. `git pull --rebase`
2. if git says "you have uncommitted changes", do a `git stash` before and try 1. again
3. `git stash pop` if you had to do the git stash to recover your pending changes

File: bundles/org.openhab.ui/web/src/pages/settings/rules/script/blockly-editor.vue (hunk 1 above)
Comment (stefan-hoehn) [reply bundles/org.openhab.ui/web/src/pages/settings/rules/script/blockly-editor.vue]: done

**Comment by stefan-hoehn** [comment]
- I had committed one proposed one change via github
- then fetched upstream to my fork, and pulled it locally which resulted into a merge
- I added my changes / format changes (during the merge to avoid just another commit)
- I committed the change and pushed it

Now I have 6 commits. Is this normal or did I do something wrong?

**Comment by ghys** [comment]
merged after finishing this PR following @stefan-hoehn's request.

**Comment by stefan-hoehn** [comment]
Thanks, Yannick

**Pull request merged**